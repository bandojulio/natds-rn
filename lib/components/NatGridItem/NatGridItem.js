import React from 'react';
import PropTypes from 'prop-types';
import { TouchableWithoutFeedback } from 'react-native';

import {
  Thumb, Item, Description, Title,
} from './NatGridItem.styles';

export const NatGridItem = ({
  source, square, onPress, children,
}) => (
  <Item square={square} accessible accessibilityLabel={`${children}`}>
    <TouchableWithoutFeedback onPress={onPress}>
      <Thumb
        accessibilityRole="image"
        square={square}
        source={{ uri: source }}
        accessibilityLabel={`${children}`}
      />
    </TouchableWithoutFeedback>
    <Description
      square={square}
      accessibilityLabel={`${children}`}
    >
      <Title>{children}</Title>
    </Description>
  </Item>
);

NatGridItem.propTypes = {
  source: PropTypes.string.isRequired,
  square: PropTypes.bool,
  onPress: PropTypes.func,
  children: PropTypes.string,
};

NatGridItem.defaultProps = {
  onPress: PropTypes.func,
  children: 'TÃ­tulo da imagem',
  square: false,
};
