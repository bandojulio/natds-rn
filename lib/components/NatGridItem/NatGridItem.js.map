{"version":3,"sources":["NatGridItem.js"],"names":["NatGridItem","source","sizeMedium","onPress","children","uri","propTypes","PropTypes","string","func","bool","isRequired","defaultProps"],"mappings":"oJAAA,4B;AACA,qC;AACA;;AAEA,iD;;;;AAIO,GAAMA,iCAAc,QAAdA,YAAc;AACzBC,MADyB,MACzBA,MADyB,CACjBC,UADiB,MACjBA,UADiB,CACLC,OADK,MACLA,OADK,CACIC,QADJ,MACIA,QADJ;;AAGzB,8BAAC,iBAAD,EAAM,WAAYF,UAAlB;AACE,8BAAC,qCAAD,EAA0B,QAASC,OAAnC;AACE,8BAAC,kBAAD;AACE,kBAAkB,OADpB;AAEE,WAAYD,UAFd;AAGE,OAAQ,CAAEG,IAAKJ,MAAP,CAHV;AAIE,sBAAuBG,QAJzB,iDADF,CADF;;;AASE,8BAAC,wBAAD;AACE,WAAYF,UADd;AAEE,sBAAuBE,QAFzB;;AAIE,8BAAC,kBAAD,kDAAQA,QAAR,CAJF,CATF,CAHyB,GAApB;;;;;AAqBPJ,YAAYM,SAAZ,CAAwB;AACtBF,SAAUG,oBAAUC,MADE;AAEtBL,QAASI,oBAAUE,IAFG;AAGtBP,WAAYK,oBAAUG,IAHA;AAItBT,OAAQM,oBAAUC,MAAV,CAAiBG,UAJH,CAAxB;;;AAOAX,YAAYY,YAAZ,CAA2B;AACzBR,SAAU,kBADe;AAEzBD,QAASI,oBAAUE,IAFM;AAGzBP,WAAY,KAHa,CAA3B","file":"NatGridItem.js","sourcesContent":["import React from 'react';\nimport PropTypes from 'prop-types';\nimport { TouchableWithoutFeedback } from 'react-native';\n\nimport {\n  Thumb, Item, Description, Title,\n} from './NatGridItem.styles';\n\nexport const NatGridItem = ({\n  source, sizeMedium, onPress, children,\n}) => (\n  <Item sizeMedium={sizeMedium}>\n    <TouchableWithoutFeedback onPress={onPress}>\n      <Thumb\n        accessibilityRole=\"image\"\n        sizeMedium={sizeMedium}\n        source={{ uri: source }}\n        accessibilityLabel={`${children}`}\n      />\n    </TouchableWithoutFeedback>\n    <Description\n      sizeMedium={sizeMedium}\n      accessibilityLabel={`${children}`}\n    >\n      <Title>{children}</Title>\n    </Description>\n  </Item>\n);\n\nNatGridItem.propTypes = {\n  children: PropTypes.string,\n  onPress: PropTypes.func,\n  sizeMedium: PropTypes.bool,\n  source: PropTypes.string.isRequired,\n};\n\nNatGridItem.defaultProps = {\n  children: 'TÃ­tulo da imagem',\n  onPress: PropTypes.func,\n  sizeMedium: false,\n};\n"]}