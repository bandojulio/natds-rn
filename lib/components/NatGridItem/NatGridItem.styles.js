import styled from 'styled-components/native';
import { robotoRegular } from '../../assets/theme/fonts.js';
import { colors } from '../../assets/theme/colors';

export const Thumb = styled.Image`
    width: ${props => (props.square ? 152 : 128)};
    height: ${props => (props.square ? 152 : 80)};
    border-radius: 6;
`;

export const Item = styled.View`
    margin-right: 12px;
    margin-left: 12px;
    margin-bottom: 24px;
    width: ${props => (props.square ? 152 : 128)};
    height: ${props => (props.square ? 200 : 128)};
`;

export const Description = styled.View`
    width: ${props => (props.square ? 152 : 128)};
    height: 40;
    line-height: 22;
    margin-top: 8;
`;

export const Title = styled.Text`
    flex: 1;
    font-size: 14;
    font-weight: 500;
    font-style: normal;
    text-align: center;
    color: ${colors.bw.grayDark};
    ${robotoRegular}
`;
