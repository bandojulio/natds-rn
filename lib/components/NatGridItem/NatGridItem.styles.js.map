{"version":3,"sources":["NatGridItem.styles.js"],"names":["Item","styled","View","props","sizeMedium","Thumb","Image","Description","Title","Text","colors","bw","grayDark","Platform","OS","sanfrancisco","regular","roboto"],"mappings":"2iCAAA;AACA,gD;AACA;AACA,wD;;AAEO,GAAMA,mBAAOC,iBAAOC,IAAd;AACM,sBAAUC,OAAMC,UAAN,CAAmB,EAAnB,CAAwB,CAAlC,EADN;AAEK,sBAAUD,OAAMC,UAAN,CAAmB,EAAnB,CAAwB,EAAlC,EAFL;AAGI,sBAAUD,OAAMC,UAAN,CAAmB,EAAnB,CAAwB,CAAlC,EAHJ;AAIF,sBAAUD,OAAMC,UAAN,CAAmB,GAAnB,CAAyB,GAAnC,EAJE;AAKD,sBAAUD,OAAMC,UAAN,CAAmB,GAAnB,CAAyB,GAAnC,EALC,CAAN;;;AAQA,GAAMC,qBAAQJ,iBAAOK,KAAf;AACF,sBAAUH,OAAMC,UAAN,CAAmB,GAAnB,CAAyB,GAAnC,EADE;AAED,sBAAUD,OAAMC,UAAN,CAAmB,GAAnB,CAAyB,GAAnC,EAFC,CAAN;;;;;AAOA,GAAMG,iCAAcN,iBAAOC,IAArB;AACF,sBAAUC,OAAMC,UAAN,CAAmB,GAAnB,CAAyB,GAAnC,EADE,CAAN;;;;;AAMA,GAAMI,qBAAQP,iBAAOQ,IAAf;;;;;;AAMFC,eAAOC,EAAP,CAAUC,QANR;AAOMC,sBAASC,EAAT,GAAgB,KAAjB,CAA0BC,oBAAaC,OAAvC,CAAiDC,cAAOD,OAP7D,CAAN","file":"NatGridItem.styles.js","sourcesContent":["import { Platform } from 'react-native';\nimport styled from 'styled-components/native';\nimport { roboto, sanfrancisco } from '../../assets/theme/fonts/fonts';\nimport { colors } from '../../assets/theme/colors/colors';\n\nexport const Item = styled.View`\n  margin-bottom: ${props => (props.sizeMedium ? 24 : 0)};\n  margin-right: ${props => (props.sizeMedium ? 12 : 16)};\n  margin-left: ${props => (props.sizeMedium ? 12 : 0)};\n  width: ${props => (props.sizeMedium ? 156 : 128)};\n  height: ${props => (props.sizeMedium ? 200 : 176)};\n`;\n\nexport const Thumb = styled.Image`\n  width: ${props => (props.sizeMedium ? 156 : 128)};\n  height: ${props => (props.sizeMedium ? 156 : 128)};\n  border-radius: 6;\n  margin-bottom: 8px;\n`;\n\nexport const Description = styled.View`\n  width: ${props => (props.sizeMedium ? 156 : 128)};\n  height: 40px;\n  line-height: 20;\n`;\n\nexport const Title = styled.Text`\n  flex: 1;\n  font-size: 14;\n  font-weight: 500;\n  font-style: normal;\n  text-align: center;\n  color: ${colors.bw.grayDark};\n  font-family:  ${(Platform.OS === 'ios') ? sanfrancisco.regular : roboto.regular};\n`;\n"]}