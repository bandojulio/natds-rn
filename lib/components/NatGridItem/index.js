Object.defineProperty(exports,"__esModule",{value:true});var _NatGridItem=require('./NatGridItem');Object.defineProperty(exports,'NatGridItem',{enumerable:true,get:function get(){return _NatGridItem.NatGridItem;}});
//# sourceMappingURL=index.js.map