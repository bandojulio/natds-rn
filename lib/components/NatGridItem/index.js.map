{"version":3,"sources":["index.js"],"names":["NatGridItem"],"mappings":"uMAASA,W","file":"index.js","sourcesContent":["export { NatGridItem } from './NatGridItem';\n"]}