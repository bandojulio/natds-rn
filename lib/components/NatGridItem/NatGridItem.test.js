import React from 'react';
import renderer from 'react-test-renderer';

import { NatGridItem } from './NatGridItem';
import {
  Thumb, Item, Description, Title,
} from './NatGridItem.styles';
import 'jest-styled-components/native';


describe('NatGridItem component', () => {
  it('Should render NatGridItem component ', () => {
    const griditem = renderer.create(
      <NatGridItem square onPress={jest.fn()} source="#">
        Texto
      </NatGridItem>,
    ).toJSON();
    expect(griditem).toMatchSnapshot();
  });

  describe('Properties of NatGridItem', () => {
    const press = jest.fn();

    const props = {
      source: '#',
      square: true,
      onPress: press,
      children: 'descrição',
    };

    it('Should render NatGridItem component ', () => {
      const griditem = renderer.create(<NatGridItem {...props} />).toJSON();
      expect(griditem).toMatchSnapshot();
    });
  });
});

describe('Style Thumb', () => {
  it('Thumb default', () => {
    const thumb = renderer.create(<Thumb />).toJSON();
    expect(thumb).toHaveStyleRule('width', 128);
    expect(thumb).toHaveStyleRule('height', 80);
    expect(thumb).toHaveStyleRule('border-radius', 6);
  });

  it('Style Thumb with square props', () => {
    const thumb = renderer.create(<Thumb square />).toJSON();
    expect(thumb).toHaveStyleRule('width', 152);
    expect(thumb).toHaveStyleRule('height', 152);
  });
});

describe('Style Item', () => {
  it('Item default', () => {
    const item = renderer.create(<Item />).toJSON();
    expect(item).toHaveStyleRule('margin-left', 12);
    expect(item).toHaveStyleRule('margin-right', 12);
    expect(item).toHaveStyleRule('margin-bottom', 12);
    expect(item).toHaveStyleRule('width', 128);
    expect(item).toHaveStyleRule('height', 128);
  });

  it('Item with square props', () => {
    const item = renderer.create(<Item square />).toJSON();
    expect(item).toHaveStyleRule('width', 152);
    expect(item).toHaveStyleRule('height', 200);
  });
});

describe('Style Description', () => {
  it('Description default', () => {
    const description = renderer.create(<Description />).toJSON();
    expect(description).toHaveStyleRule('width', 128);
    expect(description).toHaveStyleRule('height', 40);
    expect(description).toHaveStyleRule('margin-top', 8);
  });

  it('description with square props', () => {
    const description = renderer.create(<Description square />).toJSON();
    expect(description).toHaveStyleRule('width', 152);
  });
});

describe('Style Title', () => {
  it('Title default', () => {
    const title = renderer.create(<Title />).toJSON();
    expect(title).toHaveStyleRule('flex');
    expect(title).toHaveStyleRule('font-size', 14);
    expect(title).toHaveStyleRule('font-weight', '500');
    expect(title).toHaveStyleRule('font-style', 'normal');
    expect(title).toHaveStyleRule('font-align');
    expect(title).toHaveStyleRule('color', '#777777');
    expect(title).toHaveStyleRule('font-family', 'Roboto');
  });
});
