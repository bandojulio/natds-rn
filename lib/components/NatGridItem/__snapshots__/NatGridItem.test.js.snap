// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NatGridItem component Properties of NatGridItem Should render NatGridItem component  1`] = `
<View
  accessibilityLabel="descrição"
  accessible={true}
  square={true}
  style={
    Array [
      Object {
        "height": 200,
        "marginBottom": 24,
        "marginLeft": 12,
        "marginRight": 12,
        "width": 152,
      },
    ]
  }
>
  <Image
    accessibilityLabel="descrição"
    accessibilityRole="image"
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "uri": "#",
      }
    }
    square={true}
    style={
      Array [
        Object {
          "borderRadius": 6,
          "height": 152,
          "marginBottom": 8,
          "width": 152,
        },
      ]
    }
  />
  <View
    accessibilityLabel="descrição"
    square={true}
    style={
      Array [
        Object {
          "height": 40,
          "lineHeight": 22,
          "width": 152,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#777777",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontFamily": "Roboto-Regular",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "500",
            "textAlign": "center",
          },
        ]
      }
    >
      descrição
    </Text>
  </View>
</View>
`;

exports[`NatGridItem component Should render NatGridItem component  1`] = `
<View
  accessibilityLabel="Texto"
  accessible={true}
  square={true}
  style={
    Array [
      Object {
        "height": 200,
        "marginBottom": 24,
        "marginLeft": 12,
        "marginRight": 12,
        "width": 152,
      },
    ]
  }
>
  <Image
    accessibilityLabel="Texto"
    accessibilityRole="image"
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "uri": "#",
      }
    }
    square={true}
    style={
      Array [
        Object {
          "borderRadius": 6,
          "height": 152,
          "marginBottom": 8,
          "width": 152,
        },
      ]
    }
  />
  <View
    accessibilityLabel="Texto"
    square={true}
    style={
      Array [
        Object {
          "height": 40,
          "lineHeight": 22,
          "width": 152,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#777777",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontFamily": "Roboto-Regular",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "500",
            "textAlign": "center",
          },
        ]
      }
    >
      Texto
    </Text>
  </View>
</View>
`;
